From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david.llewellyn-jones@jolla.com>
Date: Fri, 16 Oct 2020 14:41:10 +0300
Subject: [PATCH] [sailfishos][gecko] Include XUL videocontrols reflow code on
 Sailfish OS

The desktop browser videocontrol UI is implemented in HTML, whereas on
Sailfish OS (and Android) XUL content is still needed. This change
ensure the XUL code used on Android is also applied for Sailfish OS.

The XUL code is also removed for Android in ESR61. See bug 1510384 and
the linked patch.

https://bugzilla.mozilla.org/show_bug.cgi?id=1510384

https://hg.mozilla.org/mozreview/gecko/rev/9e6a5f9fbde7bd3cf590bc3f8e0736fce170361d
---
 layout/generic/nsVideoFrame.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/layout/generic/nsVideoFrame.cpp b/layout/generic/nsVideoFrame.cpp
index bc9c1a4678dc..071e8ad5a5b2 100644
--- a/layout/generic/nsVideoFrame.cpp
+++ b/layout/generic/nsVideoFrame.cpp
@@ -318,7 +318,7 @@ void nsVideoFrame::Reflow(nsPresContext* aPresContext, ReflowOutput& aMetrics,
 
 // Android still uses XUL media controls & hence needs this XUL-friendly
 // custom reflow code. This will go away in bug 1310907.
-#ifdef ANDROID
+#if defined(ANDROID) || defined(MOZ_EMBEDLITE)
     } else if (child->GetContent() == mVideoControls) {
       // Reflow the video controls frame.
       nsBoxLayoutState boxState(PresContext(), aReflowInput.mRenderingContext);
@@ -327,7 +327,7 @@ void nsVideoFrame::Reflow(nsPresContext* aPresContext, ReflowOutput& aMetrics,
           nsRect(borderPadding.left, borderPadding.top,
                  aReflowInput.ComputedWidth(), aReflowInput.ComputedHeight()));
 
-#endif  // ANDROID
+#endif  // ANDROID || MOZ_EMBEDLITE
     } else if (child->GetContent() == mCaptionDiv ||
                child->GetContent() == mVideoControls) {
       // Reflow the caption and control bar frames.
@@ -566,13 +566,13 @@ LogicalSize nsVideoFrame::ComputeSize(
 // When in no video scenario, it should fall back to inherited method.
 // We keep old codepath here since Android still uses XUL media controls.
 // This will go away in bug 1310907.
-#ifndef ANDROID
+#if !defined(ANDROID) && !defined(MOZ_EMBEDLITE)
   if (!HasVideoElement()) {
     return nsContainerFrame::ComputeSize(aRenderingContext, aWM, aCBSize,
                                          aAvailableISize, aMargin, aBorder,
                                          aPadding, aFlags);
   }
-#endif  // ANDROID
+#endif  // !ANDROID && !MOZ_EMBEDLITE
 
   nsSize size = GetVideoIntrinsicSize(aRenderingContext);
 
@@ -670,7 +670,7 @@ nsSize nsVideoFrame::GetVideoIntrinsicSize(gfxContext* aRenderingContext) {
 
 // All media controls have been converted to HTML except Android. Hence
 // we keep this codepath for Android until removal in bug 1310907.
-#ifdef ANDROID
+#if defined(ANDROID) || defined(MOZ_EMBEDLITE)
   if (!HasVideoElement()) {
     if (!mFrames.FirstChild()) {
       return nsSize(0, 0);
@@ -681,7 +681,7 @@ nsSize nsVideoFrame::GetVideoIntrinsicSize(gfxContext* aRenderingContext) {
     nscoord prefHeight = mFrames.LastChild()->GetXULPrefSize(boxState).height;
     return nsSize(nsPresContext::CSSPixelsToAppUnits(size.width), prefHeight);
   }
-#endif  // ANDROID
+#endif  // ANDROID || MOZ_EMBEDLITE
 
   HTMLVideoElement* element = static_cast<HTMLVideoElement*>(GetContent());
   if (NS_FAILED(element->GetVideoSize(&size)) && ShouldDisplayPoster()) {
-- 
2.26.2

